src/fsharp/DotNetFrameworkDependencies.fs

```diff
@@ -0,0 +1,247 @@
+                        // System.Private.CoreLib doesn't load with reflection
+                        if reference.Name <> "System.Private.CoreLib" then
+                            traverseDependencies reference.Name
+                with e -> ()
+
+        assemblyReferences |> List.iter(traverseDependencies)
+        assemblies
+
+    // This list is the default set of references for "non-project" files. 
+    //
+    // These DLLs are
+    //    (a) included in the environment used for all .fsx files (see service.fs)
+    //    (b) included in environment for files 'orphaned' from a project context
+    //            -- for orphaned files (files in VS without a project context)
+    //            -- for files given on a command line without --noframework set
+    let getDesktopDefaultReferences useFsiAuxLib = [
+        yield "mscorlib"
+        yield "System"
+        yield "System.Xml"
+        yield "System.Runtime.Remoting"
+        yield "System.Runtime.Serialization.Formatters.Soap"
+        yield "System.Data"
+        yield "System.Drawing"
+        yield "System.Core"
+        yield getDefaultFSharpCoreReference
+        if useFsiAuxLib then yield getFsiLibraryName
+ 
+        // always include a default reference to System.ValueTuple.dll in scripts and out-of-project sources 
+        match getDefaultSystemValueTupleReference() with
+        | None -> ()
+        | Some v -> yield v
+
+        // These are the Portable-profile and .NET Standard 1.6 dependencies of FSharp.Core.dll.  These are needed
+        // when an F# sript references an F# profile 7, 78, 259 or .NET Standard 1.6 component which in turn refers 
+        // to FSharp.Core for profile 7, 78, 259 or .NET Standard.
+        yield "netstandard"
+        yield "System.Runtime"          // lots of types
+        yield "System.Linq"             // System.Linq.Expressions.Expression<T> 
+        yield "System.Reflection"       // System.Reflection.ParameterInfo
+        yield "System.Linq.Expressions" // System.Linq.IQueryable<T>
+        yield "System.Threading.Tasks"  // valuetype [System.Threading.Tasks]System.Threading.CancellationToken
+        yield "System.IO"               //  System.IO.TextWriter
+        yield "System.Net.Requests"     //  System.Net.WebResponse etc.
+        yield "System.Collections"      // System.Collections.Generic.List<T>
+        yield "System.Runtime.Numerics" // BigInteger
+        yield "System.Threading"        // OperationCanceledException
+        yield "System.Web"
+        yield "System.Web.Services"
+        yield "System.Windows.Forms"
+        yield "System.Numerics"
+    ]
+
+    let fetchPathsForDefaultReferencesForScriptsAndOutOfProjectSources useFsiAuxLib assumeDotNetFramework =
+        let assemblies =
+            if assumeDotNetFramework then
+                getDependenciesOf (getDesktopDefaultReferences useFsiAuxLib)
+            else
+                // Coreclr supports netstandard assemblies only for now
```

Comment:
Can we expand this to also include the closure of netcoreapp? This would allow things like span in scripting

Reply:
@cartermp 
as discussed, here is using Span.

````
c:\kevinransom\visualfsharp>artifacts\toolset\dotnet\dotnet.exe artifacts\bin\fsi\Release\netcoreapp2.1\fsi.exe

Microsoft (R) F# Interactive version 10.2.3 for F# 4.5
Copyright (c) Microsoft Corporation. All Rights Reserved.

For help type #help;;

> // Learn more about F# at http://fsharp.org

open System

//[<EntryPoint>]
let main argv =
    printfn "Hello World from F#!"

    let t = Span.Empty
    0 // return an integer exit code
let _ = main [| "Hello" |];;
Hello World from F#!
val main : argv:'a -> int

````

Span can't be a top level function in fsi.  Perhaps we need to think about how that might look if we enabled it.  Or may not.